### ISSUE TITLE
KeyError: 'type' using fs.glob

### ISSUE THREAD

**Issue by lostmygithubaccount** [dask/adlfs#34]
My data is stored in an ADLS gen2 account. It is partitioned by `year=*/month=*/part-*.snappy.parquet`. 

The setup code is:

```python
STORAGE_OPTIONS = {
    'account_name': 'data4dask', 
    'account_key' : 'my_account_key'
}

protocol  = 'abfs'      # use 'adl' for Azure Data Lake Gen 1
container = 'datasets'  #

fs = fsspec.filesystem(protocol, **STORAGE_OPTIONS, container_name=container)
```

This works:

```python
files = []
for file in fs.glob('noaa/isd/year=*/month=*'):
    files += fs.ls(f'{file}/')
files = [f'{protocol}://{container}/{file}' for file in files if '2019' not in file] # issue with 2019 data - fix wip 
files[-5:]
```
and returns the paths to the last 5 files.

This would be simpler, but doesn't work:

```python
files = fs.glob('noaa/isd/year=*/month=*/*')
files = [f'{protocol}://{container}/{file}' for file in files if '2019' not in file] # issue with 2019 data - fix wip 
files[-5:]
```

Stacktrace:
```python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-21-49b9609aae79> in <module>
      1 files = []
----> 2 files = fs.glob('noaa/isd/year=*/month=*/*')
      3 files = [f'{protocol}://{container}/{file}' for file in files if '2019' not in file] # issue with 2019 data - fix wip
      4 files[-5:]

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/fsspec/spec.py in glob(self, path, **kwargs)
    446             root = ""
    447             depth = 20 if "**" in path else 1
--> 448         allpaths = self.find(root, maxdepth=depth, withdirs=True, **kwargs)
    449         pattern = (
    450             "^"

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/fsspec/spec.py in find(self, path, maxdepth, withdirs, **kwargs)
    369         # TODO: allow equivalent of -name parameter
    370         out = set()
--> 371         for path, dirs, files in self.walk(path, maxdepth, **kwargs):
    372             if withdirs:
    373                 files += dirs

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/adlfs/core.py in walk(self, path, maxdepth, **kwargs)
    506                     d,
    507                     maxdepth=(maxdepth - 1) if maxdepth is not None else None,
--> 508                     **kwargs,
    509                 ):
    510                     yield res

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/adlfs/core.py in walk(self, path, maxdepth, **kwargs)
    506                     d,
    507                     maxdepth=(maxdepth - 1) if maxdepth is not None else None,
--> 508                     **kwargs,
    509                 ):
    510                     yield res

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/adlfs/core.py in walk(self, path, maxdepth, **kwargs)
    471             name = info["name"].rstrip("/")
    472             logging.debug(
--> 473                 f"Test path with name, path, type, size:  {name}, {path}, {info['type']}, {info['size']}"
    474             )
    475             if info["type"] == "directory" and name != path and info["size"] == 0:

KeyError: 'type'
```

**Comment by lostmygithubaccount** [comment]
Similar error trying to read using dask.dataframe.read_csv:

```python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-15-99412ae0de4e> in <module>
----> 1 df = dask.delayed(dd.read_csv)(f'abfs://testing/data/isd*/*.csv', engine='pyarrow', storage_options=STORAGE_OPTIONS).compute()
      2 get_ipython().run_line_magic('time', 'df.head()')

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/dask/base.py in compute(self, **kwargs)
    163         dask.base.compute
    164         """
--> 165         (result,) = compute(self, traverse=False, **kwargs)
    166         return result
    167 

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/dask/base.py in compute(*args, **kwargs)
    434     keys = [x.__dask_keys__() for x in collections]
    435     postcomputes = [x.__dask_postcompute__() for x in collections]
--> 436     results = schedule(dsk, keys, **kwargs)
    437     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    438 

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/client.py in get(self, dsk, keys, restrictions, loose_restrictions, resources, sync, asynchronous, direct, retries, priority, fifo_timeout, actors, **kwargs)
   2574                     should_rejoin = False
   2575             try:
-> 2576                 results = self.gather(packed, asynchronous=asynchronous, direct=direct)
   2577             finally:
   2578                 for f in futures.values():

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/client.py in gather(self, futures, errors, direct, asynchronous)
   1872                 direct=direct,
   1873                 local_worker=local_worker,
-> 1874                 asynchronous=asynchronous,
   1875             )
   1876 

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/client.py in sync(self, func, asynchronous, callback_timeout, *args, **kwargs)
    767         else:
    768             return sync(
--> 769                 self.loop, func, *args, callback_timeout=callback_timeout, **kwargs
    770             )
    771 

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/utils.py in sync(loop, func, callback_timeout, *args, **kwargs)
    333     if error[0]:
    334         typ, exc, tb = error[0]
--> 335         raise exc.with_traceback(tb)
    336     else:
    337         return result[0]

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/utils.py in f()
    317             if callback_timeout is not None:
    318                 future = gen.with_timeout(timedelta(seconds=callback_timeout), future)
--> 319             result[0] = yield future
    320         except Exception as exc:
    321             error[0] = sys.exc_info()

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/tornado/gen.py in run(self)
    733 
    734                     try:
--> 735                         value = future.result()
    736                     except Exception:
    737                         exc_info = sys.exc_info()

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/client.py in _gather(self, futures, errors, direct, local_worker)
   1728                             exc = CancelledError(key)
   1729                         else:
-> 1730                             raise exception.with_traceback(traceback)
   1731                         raise exc
   1732                     if errors == "skip":

/azureml-envs/azureml_792b2d64d1e52c01b5b979a6de29c506/lib/python3.6/site-packages/dask/utils.py in apply()

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/dask/dataframe/io/csv.py in read()
    576             storage_options=storage_options,
    577             include_path_column=include_path_column,
--> 578             **kwargs
    579         )
    580 

/azureml-envs/azureml_792b2d64d1e52c01b5b979a6de29c506/lib/python3.6/site-packages/dask/dataframe/io/csv.py in read_pandas()

/azureml-envs/azureml_792b2d64d1e52c01b5b979a6de29c506/lib/python3.6/site-packages/dask/bytes/core.py in read_bytes()

/azureml-envs/azureml_792b2d64d1e52c01b5b979a6de29c506/lib/python3.6/site-packages/fsspec/core.py in get_fs_token_paths()

/azureml-envs/azureml_792b2d64d1e52c01b5b979a6de29c506/lib/python3.6/site-packages/fsspec/spec.py in glob()

/azureml-envs/azureml_792b2d64d1e52c01b5b979a6de29c506/lib/python3.6/site-packages/fsspec/spec.py in find()

/azureml-envs/azureml_792b2d64d1e52c01b5b979a6de29c506/lib/python3.6/site-packages/adlfs/core.py in walk()

/azureml-envs/azureml_792b2d64d1e52c01b5b979a6de29c506/lib/python3.6/site-packages/adlfs/core.py in walk()

KeyError: 'type'
```

**Comment by hayesgb** [comment]
I attempted to put together a minimal example from your observation.  Starting with a set of csv files, such that there are 10 CSV files in a container titled:  "testfolder":

```
['alzhiemers.csv/0.csv',
 'alzhiemers.csv/1.csv',
 'alzhiemers.csv/2.csv',
 'alzhiemers.csv/3.csv',
 'alzhiemers.csv/4.csv',
 'alzhiemers.csv/5.csv',
 'alzhiemers.csv/6.csv',
 'alzhiemers.csv/7.csv',
 'alzhiemers.csv/8.csv',
 'alzhiemers.csv/9.csv']
```

the following behaves as expected for me:

```
from adlfs import AzureBlobFileSystem
storage_options={
    'account_name': 'account_name',
    'account_key': 'account_key',
    'container_name': "testfolder"
}
account_name='account_name'
account_key='account_key'
container='testfolder'

fs = AzureBlobFileSystem(account_name=account_name,
                        account_key=account_key,
                        container_name=container)
```
![image](https://user-images.githubusercontent.com/12595382/72436925-33e49900-3767-11ea-85d5-ef65ea83a6e2.png)

The same works if I follow your convention.  Can you confirm that you're using the latest version of adlfs?  There was a bug in 0.1.4.

Also, to instantiate the filesystem as you do above, you should be including the container_name in the **STORAGE_OPTIONS dictionary.  Finally, when reading from csv files, the engine declaration is not needed.

**Comment by lassevalentini** [comment]
I have experienced this temporarily as well. (adlfs==0.1.5, azure-storage-blob==2.1.0)

I uploaded a large partitioned parquet file to a container, and for some reason no type information were returned on at least one of the folders in the parquet "file".

My guess would be that the following [piece of code](https://github.com/dask/adlfs/commit/ef760e2765da064e7795acaf6ebda0c7980e5e92) could handle it.

Waiting for a while (and possibly cloning the folder with a new name) resolved the issue for me, so i would guess this is simply the Azure blob service not returning any type information temporarily.
